File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/CrossLineAnnotation.swift
//
//  CrossLineAnnotation.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class CrossLineAnnotation: TKExamplesExampleViewController {
    
    let chart = TKChart()
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)

        self.addOption("cross lines", action: crossLine)
        self.addOption("horizontal line", action: horizontalLine)
        self.addOption("vertical line", action: verticalLine)
        self.addOption("disable lines", action: disableLines)
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
    
        chart.frame = self.view.bounds
        chart.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(chart)
        
        for _ in 0..<2 {
            var points = [TKChartDataPoint]()
            for _ in 0..<20 {
                points.append(TKChartDataPoint(x: Int(arc4random() % (1450)), y: Int(arc4random() % (150))))
            }
            chart.addSeries(TKChartScatterSeries(items: points))
        }
        
        // Add a cross line annotation
        chart.addAnnotation(TKChartCrossLineAnnotation(x: 900, y: 60, forSeries: chart.series[0]))
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    //MARK: Events
    
    func crossLine()
    {
        let a = chart.annotations[0] as! TKChartCrossLineAnnotation
        a.style.verticalLineStroke = TKStroke(color: UIColor.blackColor())
        a.style.horizontalLineStroke = TKStroke(color: UIColor.blackColor())
        a.style.pointShape = TKPredefinedShape(type: TKShapeType.Circle, andSize: CGSizeMake(4, 4))
        chart.updateAnnotations()
    }
    
    func horizontalLine()
    {
        let a = chart.annotations[0] as! TKChartCrossLineAnnotation
        a.style.verticalLineStroke = nil
        a.style.horizontalLineStroke = TKStroke(color: UIColor.blackColor())
        a.style.pointShape = TKPredefinedShape(type: TKShapeType.Circle, andSize: CGSizeMake(4, 4))
        chart.updateAnnotations()
    }
    
    func verticalLine()
    {
        let a = chart.annotations[0] as! TKChartCrossLineAnnotation
        a.style.verticalLineStroke = TKStroke(color: UIColor.blackColor())
        a.style.horizontalLineStroke = nil
        a.style.pointShape = TKPredefinedShape(type: TKShapeType.Circle, andSize: CGSizeMake(4, 4))
        chart.updateAnnotations()
    }
    
    func disableLines()
    {
        let a = chart.annotations[0] as! TKChartCrossLineAnnotation
        a.style.verticalLineStroke = nil
        a.style.horizontalLineStroke = nil
        a.style.pointShape = TKPredefinedShape(type: TKShapeType.TriangleUp, andSize: CGSizeMake(20, 20))
        chart.updateAnnotations()
    }
}

File: Examples/Examples/Docs/Annotations/CrossLineAnnotation.cs
using System;
using System.Collections.Generic;
using System.Drawing;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	[Register("CrossLineAnnotation")]
	public class CrossLineAnnotation: XamarinExampleViewController
	{
		TKChart chart;

		public override void ViewDidLoad ()
		{
			this.AddOption ("cross lines", crossLines);
			this.AddOption ("horizontal line", horizontalLine);
			this.AddOption ("vertical line", verticalLine);
			this.AddOption ("disable lines", disableLines);

			base.ViewDidLoad ();

			chart = new TKChart (this.View.Bounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.View.AddSubview (chart);

			Random r = new Random ();
			for (int i = 0; i < 2; i++) {
				List<TKChartDataPoint> list = new List<TKChartDataPoint> ();
				for (int j = 0; j < 20; j++) {
					list.Add (new TKChartDataPoint(new NSNumber(r.Next() % 1450), new NSNumber(r.Next() % 150)));
				}
				chart.AddSeries (new TKChartScatterSeries (list.ToArray()));
			}

			// >> chart-line-stroke-cs
			chart.XAxis.Style.LineStroke = new TKStroke (UIColor.Black);
			// << chart-line-stroke-cs

			// >> chart-cross-annotation-cs
			chart.AddAnnotation (new TKChartCrossLineAnnotation(new NSNumber(900), new NSNumber(60), chart.Series[0]));
			// << chart-cross-annotation-cs
		}

		public void crossLines()
		{
			TKChartCrossLineAnnotation a = (TKChartCrossLineAnnotation)chart.Annotations [0];
			a.Style.VerticalLineStroke = new TKStroke (UIColor.Black);
			a.Style.HorizontalLineStroke = new TKStroke (UIColor.Black);
			a.Style.PointShape = new TKPredefinedShape(TKShapeType.Circle, new SizeF(4, 4));
			chart.UpdateAnnotations();
		}

		public void horizontalLine()
		{
			TKChartCrossLineAnnotation a = (TKChartCrossLineAnnotation)chart.Annotations [0];
			a.Style.VerticalLineStroke = null;
			a.Style.HorizontalLineStroke = new TKStroke (UIColor.Black);
			a.Style.PointShape = new TKPredefinedShape (TKShapeType.Circle, new SizeF (4, 4));
			chart.UpdateAnnotations ();
		}

		public void verticalLine()
		{
			TKChartCrossLineAnnotation a = (TKChartCrossLineAnnotation)chart.Annotations [0];
			a.Style.VerticalLineStroke = new TKStroke (UIColor.Black);
			a.Style.HorizontalLineStroke = null;
			a.Style.PointShape = new TKPredefinedShape (TKShapeType.Circle, new SizeF (4, 4));
			chart.UpdateAnnotations ();
		}

		public void disableLines()
		{
			TKChartCrossLineAnnotation a = (TKChartCrossLineAnnotation)chart.Annotations [0];
			a.Style.VerticalLineStroke = null;
			a.Style.HorizontalLineStroke = null;
			a.Style.PointShape = new TKPredefinedShape (TKShapeType.TriangleUp, new SizeF (20, 20));
			chart.UpdateAnnotations ();
		}
	}
}

